js/compiler/compiler.go

```diff
@@ -207,8 +208,14 @@ func (c *Compiler) compileImpl(
 ) (*goja.Program, string, error) {
 	code := src
 	state := compilationState{srcMap: srcMap, compiler: c, wrapped: wrap}
-	if wrap { // the lines in the sourcemap (if available) will be fixed by increaseMappingsByOne
-		code = "(function(module, exports){\n" + code + "\n})\n"
+	if wrap {
+		if strings.Contains(code, "//# sourceMappingURL=") {
+			// the lines in the sourcemap (if available) will be fixed by increaseMappingsByOne
+			code = "(function(module, exports){\n" + code + "\n})\n"
```

Comment:
Really just a thought, and not blocking, but I had to squint to figure out this line was different from the one in the `else` branch (it adds a `\n` before we append the code value in the function body.

Maybe setting a variable to `""` as a default, updating it in this `if` statement, and using it in both cases after the condition could avoid such squinting in the future?

```go
var conditionalNewLine string = ""

if strings.Contains(code, "//# sourceMappingURL=") {
			// the lines in the sourcemap (if available) will be fixed by increaseMappingsByOne
			conditionalNewLine = "\n"
}

code = "(function(module, exports){" + conditionalNewLine + code + "\n})\n"
```

Or something along those lines, as I'm not necessarily super happy with the above as is 🙂